### PR TITLE
Admin Menu: Remove "Links" menu

### PR DESCRIPTION
**Pull request by mmtr**
Fixes https://github.com/Automattic/wp-calypso/issues/48177

#### Changes proposed in this Pull Request:
Removes the "Links Manager" menu from the new unified menu since its usage in Core is [highly discouraged](https://core.trac.wordpress.org/ticket/21307#comment:73) and was deprecated very long ago. 

#### Jetpack product discussion
N/A.

#### Does this pull request change what data or activity we track or use?
No.

#### Testing instructions:
* Apply D58364-code to your WP.com sandbox.
* Sandbox the API and a WP.com simple site.
* Load that simple site in either Calypso or WP Admin.
* Make sure the "Links" menu is gone.
* Make sure you can still access it directly via `/wp-admin/link-manager.php`.
* Install Jetpack beta on a WoA site and switch to the branch of this PR.
* Load that WoA site in either Calypso or WP Admin.
* Make sure the "Links" menu is gone.
* Make sure you can still access it directly via `/wp-admin/link-manager.php`.

### CODE REVIEW

File: projects/plugins/jetpack/changelog/remove-admin-menu-links-manager [hunk 1]
```diff
@@ -0,0 +1,4 @@
+Significance: patch
```
Comment (tjcafferkey) [inline projects/plugins/jetpack/changelog/remove-admin-menu-links-manager:1]: Not a big deal but I'd consider this `minor` seeing as we're not really fixing a bug, but rather deprecating some existing functionality.

**Comment by tjcafferkey** [comment]
Looks good to me, and works well on Simple sites after applying the diff. Just waiting for the branch to be available in Beta before confirming Atomic.

**Review by tjcafferkey — APPROVED** [review]
Works as expected on both Simple and Atomic sites. LGTM permitting checks pass 😄

**Review by jeherve — APPROVED** [review]

**Comment by getdave** [comment]
Thanks for the fix. Worth keeping an eye on whether this will need a hotfix via WPCOMSH given that this won't go out until April.

**Comment by mmtr** [comment]
> Thanks for the fix. Worth keeping an eye on whether this will need a hotfix via WPCOMSH given that this won't go out until April.

I think it's fine for now. If we start receiving reports from customers, we can implement the hotfix.

**Comment by mmtr** [comment]
Deployed to WP.com in r222546-wpcom.

**Comment by jeherve** [comment]
Cherry-picked to `jetpack/branch-9.5` in 636e74d7f161994e4ccfe1e42b8870f8579ede8e

**Pull request merged**